ParkingLotApiTest/ControllerTest/OrdersControllerTest.cs

```diff
@@ -0,0 +1,149 @@
+using System.Threading.Tasks;
+using ParkingLotApi;
+using Xunit;
+using System.Net;
+using System.Linq;
+using ParkingLotApi.Dto;
+
+namespace ParkingLotApiTest.ControllerTest
+{
+    [Collection("ParkingLotTest")]
+    public class OrdersControllerTest : TestBase
+    {
+        public OrdersControllerTest(CustomWebApplicationFactory<Startup> factory) : base(factory)
+        {
+        }
+
+        [Fact]
+        public async Task Should_Add_Order_Successfully_Given_Space_Avaliable_And_Parking_Lot_Existed()
+        {
+            //given
+            var client = GetClient();
+            var parkingLot = GenerateParkingLotDtoInstance();
+            var requestBody = Serialize(parkingLot);
+            var parkingLotPostResponse = await client.PostAsync("/ParkingLots", requestBody);
+            var car = new CarDto() { PlateNumber = "N95024" };
+            var carRequestBody = Serialize(car);
+            //when
+            var orderPostResponse = await client.PostAsync($"{parkingLotPostResponse.Headers.Location}/Orders", carRequestBody);
+
+            //then
+            Assert.Equal(HttpStatusCode.Created, orderPostResponse.StatusCode);
+
+            var responseBody = await DeSerializeResponseAsync<CarDto>(orderPostResponse);
+            Assert.Equal(car.PlateNumber, responseBody.PlateNumber);
+
+            var dbContext = GetContext();
+            Assert.Equal(1, dbContext.Orders.Count());
+        }
+
+        [Fact]
+        public async Task Should_Return_Not_Found_Given_Parking_Lot_Not_Existed()
+        {
+            //given
+            var client = GetClient();
+            var parkingLot = GenerateParkingLotDtoInstance();
+            var requestBody = Serialize(parkingLot);
+            var parkingLotPostResponse = await client.PostAsync("/ParkingLots", requestBody);
+            var car = new CarDto() { PlateNumber = "N95024" };
+            var carRequestBody = Serialize(car);
+            //when
+            var orderPostResponse = await client.PostAsync($"{parkingLotPostResponse.Headers.Location}200/Orders", carRequestBody);
+
+            //then
+            Assert.Equal(HttpStatusCode.NotFound, orderPostResponse.StatusCode);
+        }
+
+        [Fact]
+        public async Task Should_Return_The_Parking_Lot_IS_Full_Given_Parking_Lot_is_Full()
+        {
```

Comment:
Could add parking lot name/plate number null check cases due to previous discussion.